packages/next-app/components/DiscordHandler.tsx

```diff
@@ -0,0 +1,117 @@
+import { FC } from "react";
+import { css } from "@emotion/react";
+import { signIn, signOut, useSession } from "next-auth/react";
+import Image from "next/image";
+
+const DiscordHandler: FC = () => {
+	const { data: session } = useSession();
+
+	const avatarLoader = ({ src, width }) => {
+		return `${src}?w=${width}`;
+	};
+
+	return (
+		<div>
+			{session?.authProvider === "discord" && (
+				<div css={styles.username}>
+					<p>{session?.user?.name}</p>
+				</div>
+			)}
```

Comment:
Can we avoid nested ternary in this block below 👇  and use something like you have as above ☝️